src/segment.rs

```diff
@@ -302,6 +302,45 @@ impl Segment {
         unsafe { self.mmap.as_slice() }
     }
 
+    /// Unloads the mmap by replacing it with an empty one. This is used to release the underlying file
+    /// used by the mmap.
+    pub fn unload_mmap(&mut self) -> Result<()> {
+        let _ = std::mem::replace(&mut self.mmap, MmapViewSync::anonymous(0)?);
+        Ok(())
+    }
+
+    /// Reloads the mmap by mmaping the file again.
+    pub fn reload_mmap(&mut self) -> Result<()> {
+        let mut retries = 0;
+        const MAX_RETRIES: usize = 5;
+
+        while retries < MAX_RETRIES {
+            let result: Result<()> = (|| {
+                let file = OpenOptions::new()
+                    .read(true)
+                    .write(true)
+                    .create(false)
+                    .open(&self.path)?;
+                let capacity = (file.metadata()?.len() as usize) & !7;
+                self.mmap = MmapViewSync::from_file(&file, 0, capacity)?;
+                Ok(())
+            })();
+
+            match result {
+                Ok(_) => break,
+                Err(e) if retries == MAX_RETRIES - 1 => {
```

Comment:
One additional thought is that it might be actually reasonable to include the whole reload operation within the `reload` function as it doesn't seem like an operation that should be freely available on the public API of the object, because it's there only for this specific case.